src/Files.App/BaseLayout.cs

```diff
@@ -527,15 +497,15 @@ public void SetSelectedItemsOnNavigation()
             }
         }
 
-        private CancellationTokenSource groupingCancellationToken;
+        private CancellationTokenSource? groupingCancellationToken;
 
-        private async void FolderSettings_GroupOptionPreferenceUpdated(object sender, GroupOption e)
+        private async void FolderSettings_GroupOptionPreferenceUpdated(object? sender, GroupOption e)
         {
             // Two or more of these running at the same time will cause a crash, so cancel the previous one before beginning
             groupingCancellationToken?.Cancel();
             groupingCancellationToken = new CancellationTokenSource();
             var token = groupingCancellationToken.Token;
-            await ParentShellPageInstance.FilesystemViewModel.GroupOptionsUpdated(token);
+            await ParentShellPageInstance!.FilesystemViewModel.GroupOptionsUpdated(token);
```

Comment:
Similar to the comment below: is it correct to use "!" to get remove the nullable warning here? Perhaps we should add a null check instead or maybe ParentShellPageInstance should not be nullable in the first place.

Reply:
I setted `ParentShellPageInstance` as nullable because the default value assigned to it is null (line @79).
Given this I think the null check is the best option. Anyway, if you think that it would be better to set it as non-nullable I'll change that

Reply:
Honestly not sure. ParentShellPageInstance can indeed be null since it's not assigned in the constructor. On the other hand we use ParentShellPageInstance without null check pretty much everywhere because we assume that OnNavigatedTo() is called before everything else.

Reply:
I added `!` because of that,  I assumed that `OnNavigatedTo()` would be called always, so I didn't add the null checks because I thought it would be less performant

Reply:
Nevermind, for now the current solution is fine 👍

Reply:
At least for this PR I'd keep the current solution 👍. We can work on proper null checks in a future one.